core/src/main/java/org/apache/accumulo/core/util/cleaner/CleanerUtil.java

```diff
@@ -107,10 +107,14 @@ public static Cleanable shutdownThreadPoolExecutor(Object tpe, Logger log) {
     requireNonNull(tpe);
     requireNonNull(log);
     return CLEANER.register(tpe, () -> {
+      ThreadPoolExecutor pool = (ThreadPoolExecutor) tpe;
```

Comment:
Why cast here? Why not just make the type more restrictive in the method signature 4 lines above?